### PR TITLE
Baseline speckit

### PR DESCRIPTION
**Pull request by Jylhis**
Update gitinore

Ratify constitution and document baseline state

This commit formalizes the Jotain project's architectural principles and validates the existing codebase against a comprehensive baseline specification.

### Constitution (Version 1.0.0)
- **Seven core principles**: Nix-only package management, modular Elisp architecture, reproducibility, testing discipline, platform portability, performance optimization, and simplicity
- **Technical constraints**: Emacs 30+ PGTK, flake-based builds, completion stack (Vertico/Corfu/Consult/Orderless/Embark), Eglot LSP client, dual-theme system (doom-nord-light / nord)
- **Development workflow**: Conventional commits, CI quality gates (formatting, smoke/fast/full test suites, NMT module tests, optional VM runtime test)
- **Governance**: Constitution is authoritative; amendments require version bump and review; compliance checked in all PRs

### Baseline Specification (Jotain 001-baseline)
- **Specification**: 367-line document covering 6 user stories (P1: install+launch, coding with LSP; P2: Git, Org-mode; P3: themes, multi-platform), 19 functional requirements, measurable success criteria
- **Data Model**: 8 entity types (Elisp Module, Package Declaration, Runtime Dependency, Home Manager Option, Platform, Theme, Test Suite, Platform) with relationships
- **Contract**: Home Manager module interface documented with 4 options (enable, enableDaemon, includeRuntimeDeps, extraPackages) and artifact outputs
- **Implementation Plan**: 136-line technical context, project structure overview, all 7 constitution principles validated against baseline
- **Research**: Decision rationale for Emacs 30+/PGTK, Eglot, Vertico/Corfu stack, Nix-only package management, auto-dependency extraction, platform detection, dual-theme system; identified gaps (Windows support correctly out-of-scope, 1Password graceful failure coverage gap, startup time benchmarking needed)
- **Quickstart**: Installation instructions, verification checklist, 8 key keybindings, configuration options, troubleshooting
- **Requirements Checklist**: 42-point specification quality validation (no [NEEDS CLARIFICATION] markers, all acceptance scenarios defined, success criteria measurable)

### Test Coverage & Validation
- **test-smoke.el**: Added 4 edge case tests (font fallback, missing tree-sitter grammar, missing LSP server, startup time measurement)
- **test-auth-source-1password.el**: Added graceful failure test for missing `op` CLI (FR-019 gap)
- **test-uncovered-modules.el**: 9 new smoke tests for previously untested modules (dashboard, help, per-project, collaboration, systems)
- **test-platform.el**: Added `platform-cond` macro test coverage

### Build & Documentation
- **justfile**: Added `--no-link` flag to all Nix test builds (faster rebuild cycles); added 3 new NMT test cases (test-runtime-deps-enabled, test-runtime-deps-disabled, test-daemon-disabled) bringing total NMT coverage to 5 cases
- **Constitution.md**: Ratified and versioned (1.0.0); replaced template with 260-line substantive document
- **CLAUDE.md**: Updated with Active Technologies and Recent Changes sections extracted from baseline plan
- **Shell scripts** (.specify/scripts/bash/): Reformatted to consistent 2-space indentation

### Tasks Documentation
305-line task breakdown organized by phase and user story:
- Phase 1: Structure validation (5 tasks)
- Phase 2: Foundational build system (13 critical tasks, gate for user story validation)
- Phases 3-8: User story validation (6 stories × ~7 tasks each = 42 tasks)
- Phase 9: Gap-filling and polish (11 tasks including new test coverage)
- Execution order, parallel opportunities, MVP strategy, and multi-agent implementation guidance

## Rationale

This baseline serves three purposes:

1. **Specification**: Documents current state unambiguously, enabling future features to be specified incrementally against a known-good foundation
2. **Validation**: All 7 constitution principles verified; 68 executable tasks validate existing functionality against spec; gaps identified and addressed
3. **Governance**: Constitution establishes architectural authority; future PRs must verify compliance; clear amendment process for principle changes

The spec ratifies existing decisions (Emacs 30+/PGTK, Eglot, Vertico stack, Nix-only packaging) without imposing new requirements. No breaking changes; code is unchanged except for documentation, test coverage expansion, and formatting.

### CODE REVIEW

**Comment by Jylhis** [comment]
* **#29** <a href="https://app.graphite.com/github/pr/Jylhis/jotain/29?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/Jylhis/jotain/29?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: specs/001-baseline/spec.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+  is provided
+
+**Out of scope**:
+- User-defined module systems or plugin architectures beyond
+  `extraPackages` and `custom.el`
+- Email clients (mu4e is available on Linux but not a core feature)
+- Window manager integration (EXWM is referenced in platform code
+  but not a supported workflow)
+- Alternative completion stacks (Helm, Ivy) or LSP clients (lsp-mode)
+- Package installation at runtime from any source
```
Comment (Jylhis) [inline specs/001-baseline/spec.md]: No persistent package installation. User can still load packages adhoc but these are not persistent across sessions

File: specs/001-baseline/spec.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
+# Feature Specification: Jotain Baseline
+
+**Feature Branch**: `001-baseline`
+**Created**: 2026-02-22
+**Status**: Validated
+**Input**: Baseline specification documenting the current state of the Jotain modular Emacs distribution
+
+## Scope
+
+Jotain is an opinionated, curated Emacs distribution. It is NOT a
```
Comment (Jylhis) [inline specs/001-baseline/spec.md:10]: For generalist software engineer

File: specs/001-baseline/spec.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+3. **Given** auto-dark is enabled and the OS switches to dark mode,
+   **When** Emacs detects the change,
+   **Then** the theme automatically switches to the dark variant.
+
+---
+
+### User Story 6 - Use Jotain Across Platforms (Priority: P3)
+
+A developer uses Jotain on Linux (primary), macOS (secondary), and
```
Comment (Jylhis) [inline specs/001-baseline/spec.md:220]: GUI and terminal

File: specs/001-baseline/spec.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+- **SC-001**: A new user can install Jotain via Home Manager and
+  have a working Emacs environment after a single `nixos-rebuild`
+  or `home-manager switch` command.
+- **SC-002**: Emacs starts without errors or warnings on all
+  supported platforms (Linux, macOS, Android).
+- **SC-003**: All smoke tests pass in under 1 second, all fast
+  tests in under 5 seconds.
+- **SC-004**: The Emacs daemon starts and accepts `emacsclient`
+  connections within 10 seconds of service activation.
+- **SC-005**: At least 7 LSP servers are available and functional
```
Comment (Jylhis) [inline specs/001-baseline/spec.md]: The number is irrelevant, but at least minimal amount to confidently say that things work. We only want to have also for supported languages

File: specs/001-baseline/data-model.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+| emacs-theme | symbol | Actual Emacs theme name |
+| variable | symbol | `jotain-theme-{id}` |
+| keybinding | string | Toggle keybinding |
+
+**Current themes**:
+
+| ID | Emacs Theme | Variable | Toggle |
+|----|-------------|----------|--------|
+| light | `doom-nord-light` | `jotain-theme-light` | `C-c t` |
+| dark | `nord` | `jotain-theme-dark` | `C-c t` |
```
Comment (Jylhis) [inline specs/001-baseline/data-model.md:140]: doom-nord

**Comment by Jylhis** [comment]
Closing: targets CLAUDE.md and specs/001-baseline/ content that has since been rewritten; trivial baseline edit no longer applies.
